dotCMS/src/main/java/com/dotcms/rest/IntegrityDataRequestChecker.java

```diff
@@ -0,0 +1,188 @@
+        final String endpointId = paramsMap.get("endpoint");
+        final User loggedUser = initData.getUser();
+        final FormDataMultiPart form = new FormDataMultiPart();
+        form.field("AUTH_TOKEN",authToken);
+        form.field("REQUEST_ID", integrityDataRequestId);
+        final String url = endpoint.toURL() + "/api/integrity/getintegritydata/";
+
+        boolean processing = true;
+        while(processing) {
+            final Response response = IntegrityResource.postWithEndpointState(
+                    endpoint.getId(),
+                    url,
+                    new MediaType("application", "zip"),
+                    Entity.entity(form, form.getMediaType()));
+
+            if (response.getStatus() == HttpStatus.SC_OK) {
+                processing = false;
+
+                final String outputDir = IntegrityUtil.getIntegrityDataPath(endpoint.getId());
+                final InputStream zipFile = response.readEntity(InputStream.class);
+
+                try {
+                    IntegrityUtil.unzipFile(zipFile, outputDir);
+                } catch(Exception e) {
+                    //Special handling if the thread was interrupted
+                    if (e instanceof InterruptedException) {
+                        //Setting the process status
+                        IntegrityResource.setStatus(session, endpointId, IntegrityResource.ProcessStatus.CANCELLED, null);
+                        final String message = "Requested interruption of the integrity checking process [unzipping Integrity Data] by the user.";
+                        Logger.debug(IntegrityResource.class, message, e);
+                        throw new RuntimeException(message, e);
+                    }
+
+                    //Setting the process status
+                    IntegrityResource.setStatus(session, endpointId, IntegrityResource.ProcessStatus.ERROR, null);
+                    final String message = "Error while unzipping Integrity Data";
+                    Logger.error(IntegrityResource.class, message, e);
+                    throw new RuntimeException(message, e);
+                } finally {
+                    if (zipFile != null) {
+                        try {
+                            zipFile.close();
+                        } catch ( IOException e ) {
+                            Logger.warn(IntegrityResource.class, "Error closing zip file stream", e);
+                        }
+                    }
+                }
+
+                // set session variable
+                // call IntegrityChecker
+                boolean conflictPresent ;
+                try {
+                    HibernateUtil.startTransaction();
+                    IntegrityUtil.completeDiscardConflicts(endpointId);
+                    HibernateUtil.commitTransaction();
+
+                    HibernateUtil.startTransaction();
+                    conflictPresent = IntegrityUtil.completeCheckIntegrity(endpointId);
```

Comment:
we should annotate IntegrityUtil.completeDiscardConflicts as @WrapInTransaction

Reply:
Removed and using @WrapInTransaction annotation.